src/test/java/spoon/testing/utils/ModelTestParameterResolver.java

```diff
@@ -61,6 +69,101 @@ private Launcher createLauncher(Executable method) {
 			launcher.addInputResource(path);
 		}
 		launcher.buildModel();
+
+		// ensure that the model is valid
```

Comment:
And where would I find that test?

Reply:
`AstCheckerTest#testAvoidSetCollectionSavedOnAST` (I'd rename it and stuff it in there as a second processor, I think)

Reply:
I added it as a test, but this does not seem to detect much.

My test that hooks into the annotation, was able to detect these bugs:
- the `CtRecordPattern` where the type was not cloned
- and the broken `toMethod` implementation for `CtRecord`

(it was not able to detect the one for catch variable reference)

I reverted my fixes and ran the new `AstCheckerTest`, which was unable to find any of the above bugs.

At first, I thought it was the model, but it has all spoon classes in it? My guess is that it does not detect these bugs, because they rely on more modern language features that spoon does not use yet?